receiver/snmpreceiver/otel_metric_helper.go

```diff
@@ -174,19 +177,20 @@ func (h *otelMetricHelper) addMetricDataPoint(resourceKey string, metricName str
 		return nil, fmt.Errorf("cannot retrieve datapoints from metric '%s' as it does not currently exist", metricName)
 	}
 
-	var dps pmetric.NumberDataPointSlice
+	var dp pmetric.NumberDataPoint
 	var valueType string
 	if metricCfg.Gauge != nil {
-		dps = metric.Gauge().DataPoints()
+		dp = metric.Gauge().DataPoints().AppendEmpty()
 		valueType = metricCfg.Gauge.ValueType
 	} else {
-		dps = metric.Sum().DataPoints()
+		dp = metric.Sum().DataPoints().AppendEmpty()
+		dp.SetStartTimestamp(h.dataPointStartTime)
```

Comment:
We can use static start time for cumulative sums only. For deltas, it should be a timestamp of the previous scrape. Can the sums have delta temporality here?

Reply:
I wouldn't guess it's very common but yes, the user can configure it that way.

Are we handling this in mdatagen? I think we might be making the same mistake [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/7c16b41254a80608c7d962674f7cda9754471dd8/cmd/mdatagen/templates/metrics.go.tmpl#L374).

Just spitballing, but I think a proper solution requires a cache that can recall the most recent start time for each data point. It'd probably need to manage values be hierarchically (e.g. resource attributes -> scope attributes -> metric name -> data point attributes) and have a reasonable eviction strategy, possibly configurable. Seems like something worth standardizing. WDYT?

Reply:
> Are we handling this in mdatagen? I think we might be making the same mistake [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/7c16b41254a80608c7d962674f7cda9754471dd8/cmd/mdatagen/templates/metrics.go.tmpl#L374).

> Just spitballing, but I think a proper solution requires a cache that can recall the most recent start time for each data point. It'd probably need to manage values hierarchically (e.g. resource attributes -> scope attributes -> metric name -> data point attributes) and have a reasonable eviction strategy, possibly configurable. Seems like something worth standardizing. WDYT?

That's correct. It's not that simple. I was going to fix the issue in the metrics builder by applying what you described in the scope of https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/18062

We can leave it as is for now, but maybe capture it in a separate issue?

Reply:
Ok, makes sense. I've captured it for this receiver here: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/18064